Automattic/jetpack
# Revue Block

#### Changes proposed in this Pull Request:

* Add a new Revue block.

For this simple implementation we have decided to not use the Revue API, but simply show a subscription form, linked to a Revue user through its username, that on submit opens Revue in a new tab.

| Editor | Front End |
| - | - |
| <img width="623" alt="Screenshot 2020-02-19 at 16 02 35" src="https://user-images.githubusercontent.com/2070010/74850852-5b182280-5332-11ea-9825-fe0dd240e2cf.png"> | <img width="608" alt="Screenshot 2020-02-19 at 16 03 14" src="https://user-images.githubusercontent.com/2070010/74850862-5e131300-5332-11ea-9664-d9f1b385d58b.png"> |

**Notes**

- I've decided to reimplement (yet again) the submit button logic, copying it from the latest Gutenberg version (as of last week) that also has gradient backgrounds available, instead of using our own `SubmitButton`.
This is mostly to be as consistent as possible to Core Gutenberg, both on the edit side and on the SSR.
I recognize this is definitely not ideal, and I'd rather use a single button implementation that was as similar to Core Gutenberg's Button block.
@pablinos (since you're working on it) do you think the work in this PR could be a good starting point for a `SubmitButton` update?

- Form styles aren't typically exposed by the theme to the editor to avoid conflicts with the editor styles. For this reason, the design is slightly different between editor and front end.
I don't have a good solution for this off the top of my head, but maybe I'm missing something. 🤔 

- Since the submit opens a new tab, going back to the post and seeing the form as nothing happened felt a bit awkward to me.
Currently, 1 second after the submission, when the user is supposedly on the Revue page, the form is replaced by a "You can complete your registration in the new page." message.
This message is not permanent, and it's enough to reload the post to see the form again.
I'm not sold on the copy or the look, and I guess we could embellish them in several ways.
What do you think? (cc @davemart-in)

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* pb5gDS-jZ-p2

#### Testing instructions:






* Create an account on [Revue](https://www.getrevue.co/).
* Open the editor and insert a Revue block.
* Enter your Revue username.
* Try customizing the submit button.
* Save and view the post.
* Make sure the button show up with the same customizations in the front end.
* Fill the form with an email address (not the same as the Revue account).
* On submit you should open a Revue page with the next steps instructions.
* Check your email for a Revue email to confirm that the subscription happened (whatever happens next is in their hands).

#### Proposed changelog entry for your changes:

* Add a new Revue block.
